package/kedro_viz/models/graph.py

```diff
@@ -585,6 +597,20 @@ def __post_init__(self, data_node: DataNode):
             with dataset._fs.open(load_path, **dataset._fs_open_args_load) as fs_file:
                 self.plot = json.load(fs_file)
 
+        if data_node.is_image_node():
+            from kedro.extras.datasets.matplotlib.matplotlib_writer import (
+                MatplotlibWriter,
+            )
+
+            dataset = cast(MatplotlibWriter, dataset)
+            if not dataset._exists():

```

Comment:
Minor suggestion, but it's better practice to use the public methods where they exist. Similarly for the other occurrences in this file which this was copied from.
```suggestion
            if not dataset.exists():

```